src/main/scala/com/microsoft/hyperspace/index/dataskipping/util/ExpressionUtils.scala

```diff
@@ -108,4 +112,51 @@ object ExpressionUtils {
     override def dataType: DataType = throw new NotImplementedError
     // $COVERAGE-ON$
   }
+
+  /**
+   * Returns a normalized expression so that the indexed expression and an
+   * expression in the filter condition can be matched. For example,
+   * expressions in the filter condition can have different ExprIds for every
+   * execution, whereas the indexed expression is fixed.
+   */
+  def normalize(expr: Expression): Expression = {
+    expr.transformUp {
+      case a: AttributeReference => a.withExprId(nullExprId).withQualifier(Nil)
+      case g @ GetStructField(child, ordinal, _) => g.copy(child, ordinal, None)
+      // Undo HandleNullInputsForUDF
```

Comment:
Thanks, now it's not needed and removed.

FYI, it is an optimization rule that inserts null checks to UDF invocation, because some UDFs cannot handle null values or it's inefficient for them.